src/app/components/LoginPage/index.js

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import Failed from './Failed';
+import styles from './login.scss';
+import Header from '../NavBar/Header';
+
+let userInfo = [];
+let isRedirect = 0;
+
+const HandleRedirect = () => {
+  if (isRedirect === 1) {
+    isRedirect = 0;
+    return <Redirect to="/" />;
+  }
+  if (isRedirect === 2) {
+    isRedirect = 0;
+    return <Failed />;
+  }
+  return '';
+};
+
+class LoginPage extends Component {
+  constructor() {
+    super();
+    this.state = {
+      username: '',
+      userPassword: '',
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentDidMount() {
+    const json = localStorage.getItem('users');
+    if (json !== null) {
+      let user;
+      if (localStorage.users) {
+        user = JSON.parse(json);
+      }
+      if (user) {
+        userInfo = user;
+      }
+    }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { username, userPassword } = this.state;
+    const name = String(username);
+    const pass = String(userPassword);
+    if (name !== userInfo.username[0] || pass !== userInfo.userPassword[0]) {
+      isRedirect = 2;
+    } else {
+      isRedirect = 1;
+    }
```

Comment:
Don't use global variable.

React does not watch global variables. Even if `isRedirect` is changed, there is no guarantee that `render` function will be triggered.

Use React's state for this purpose.